src/lib/flight_tasks/tasks/AutoLineSmoothVel/FlightTaskAutoLineSmoothVel.cpp

```diff
@@ -178,53 +178,55 @@ float FlightTaskAutoLineSmoothVel::_constrainAbs(float val, float max)
 }
 
-float FlightTaskAutoLineSmoothVel::_getSpeedAtTarget(float next_target_speed) const
+float FlightTaskAutoLineSmoothVel::_getMaxXYSpeed() const
 {
-	// Compute the maximum allowed speed at the waypoint assuming that we want to
-	// connect the two lines (prev-current and current-next)
-	// with a tangent circle with constant speed and desired centripetal acceleration: a_centripetal = speed^2 / radius
-	// The circle should in theory start and end at the intersection of the lines and the waypoint's acceptance radius.
-	// This is not exactly true in reality since Navigator switches the waypoint so we have to take in account that
-	// the real acceptance radius is smaller.
-	// It can be that the next waypoint is the last one or that the drone will have to stop for some other reason
-	// so we have to make sure that the speed at the current waypoint allows to stop at the next waypoint.
-	float speed_at_target = 0.0f;
-
-	const float distance_current_next = (_target - _next_wp).xy().norm();
-	const bool waypoint_overlap = (_target - _prev_wp).xy().norm() < _target_acceptance_radius;
-	const bool yaw_align_check_pass = (_param_mpc_yaw_mode.get() != 4) || _yaw_sp_aligned;
-
-
-	if (distance_current_next > 0.001f &&
-	    !waypoint_overlap &&
-	    yaw_align_check_pass) {
-		Vector3f pos_traj;
-		pos_traj(0) = _trajectory[0].getCurrentPosition();
-		pos_traj(1) = _trajectory[1].getCurrentPosition();
-		pos_traj(2) = _trajectory[2].getCurrentPosition();
-		// Max speed between current and next
-		const float max_speed_current_next = _getMaxSpeedFromDistance(distance_current_next, next_target_speed);
-		const float alpha = acosf(Vector2f((_target - pos_traj).xy()).unit_or_zero().dot(
-						  Vector2f((_target - _next_wp).xy()).unit_or_zero()));
-		// We choose a maximum centripetal acceleration of MPC_ACC_HOR * MPC_XY_TRAJ_P to take in account
-		// that there is a jerk limit (a direct transition from line to circle is not possible)
-		// MPC_XY_TRAJ_P should be between 0 and 1.
-		float accel_tmp = _param_mpc_xy_traj_p.get() * _param_mpc_acc_hor.get();
-		float max_speed_in_turn = math::trajectory::computeMaxSpeedInWaypoint(alpha,
-					  accel_tmp,
-					  _target_acceptance_radius);
-		speed_at_target = math::min(math::min(max_speed_in_turn, max_speed_current_next), _mc_cruise_speed);
+	Vector3f pos_traj(_trajectory[0].getCurrentPosition(),
+			  _trajectory[1].getCurrentPosition(),
+			  _trajectory[2].getCurrentPosition());
+
+	math::trajectory::VehicleDynamicLimits config;
+	config.z_accept_rad = _param_nav_mc_alt_rad.get();
+	config.xy_accept_rad = _target_acceptance_radius;
+	config.max_acc_xy = _trajectory[0].getMaxAccel();
+	config.max_jerk = _trajectory[0].getMaxJerk();
+	config.max_speed_xy = _mc_cruise_speed;
+	config.max_acc_xy_radius_scale = _param_mpc_xy_traj_p.get();
+
+	Vector3f waypoints[3];
+	waypoints[0] = pos_traj;
+
+	// constrain velocity to go to the position setpoint first if our tracking is really bad
```

Comment:
can you maybe add "this is required because the avoidance sends position setpoints and not waypoints" or something similar such that the next person looking at it understands why we need to care about the position setpoint?